javacutil/src/main/java/org/checkerframework/javacutil/Pair.java

```diff
@@ -22,6 +22,70 @@ public static <V1, V2> Pair<V1, V2> of(V1 v1, V2 v2) {
     return new Pair<>(v1, v2);
   }
 
+  /**
+   * Returns a copy of this in which each element is a clone of the corresponding element of this.
+   * {@code clone()} may or may not itself make a deep copy of the elements.
+   *
+   * @param <V1> the type of the first element of the pair
+   * @param <V2> the type of the second element of the pair
+   * @param orig a pair
+   * @return a copy of {@code orig}, with all elements cloned
+   */
+  @SuppressWarnings("nullness") // generics problem with deepCopy()
+  public static <V1 extends Cloneable, V2 extends Cloneable> Pair<V1, V2> cloneElements(
+      Pair<V1, V2> orig) {
+    // Cannot modify result of super.clone() because fields are final.
+
+    V1 oldFirst = orig.first;
+    V1 newFirst = oldFirst == null ? oldFirst : CollectionUtils.clone(oldFirst);
+    V2 oldSecond = orig.second;
+    V2 newSecond = oldSecond == null ? oldSecond : CollectionUtils.clone(oldSecond);
+    return of(newFirst, newSecond);
+  }
+
+  /**
+   * Returns a deep copy of this: each element is a deep copy of the corresponding element of this.
+   *
+   * @param <V1> the type of the first element of the pair
+   * @param <V2> the type of the second element of the pair
+   * @param orig a pair
+   * @return a deep copy of {@code orig}
+   */
+  @SuppressWarnings("nullness") // generics problem with deepCopy()
+  public static <V1 extends DeepCopyable, V2 extends DeepCopyable> Pair<V1, V2> deepCopy(
+      Pair<V1, V2> orig) {
+    // Cannot modify result of super.clone() because fields are final.
+
+    V1 oldFirst = orig.first;
+    @SuppressWarnings("unchecked")
+    V1 newFirst = oldFirst == null ? oldFirst : (V1) oldFirst.deepCopy();
+    V2 oldSecond = orig.second;
+    @SuppressWarnings("unchecked")
+    V2 newSecond = oldSecond == null ? oldSecond : (V2) oldSecond.deepCopy();
+    return of(newFirst, newSecond);
+  }
+
+  /**
+   * Returns a copy, where the {@code second} element is deep: {@code first} elements are identical
+   * to the argument, and {@code} second elements are deep copies.
+   *
+   * @param <V1> the type of the first element of the pair
+   * @param <V2> the type of the second element of the pair
+   * @param orig a pair
+   * @return a copy of {@code orig}, where the second element is a deep copy
+   */
+  @SuppressWarnings("nullness") // generics problem with deepCopy()
+  public static <V1, V2 extends DeepCopyable> Pair<V1, V2> deepCopySecond(Pair<V1, V2> orig) {
```

Comment:
It seems odd to include `deepCopySecond` but not `deepCopyFirst`.

Further, I don't see any calls to this method in this PR. (So maybe this method should be removed and PR'd separately, when it is needed?)